### ISSUE TITLE
Build failure with xpdf-4.04

### ISSUE THREAD

**Issue by thesamesam** [rrthomas/libpaper#29]
I'm just filing it here in case as xpdf upstream are IIRC kind of hard to reach.

Toralf who runs some automated testing in Gentoo hit a build failure in app-text/xpdf-4.04 with libpaper-1.2.4 (I've just pushed 2.0.0) at https://bugs.gentoo.org/870217:
```
FAILED: xpdf/CMakeFiles/xpdf_objs.dir/GlobalParams.cc.o 
/usr/bin/x86_64-pc-linux-gnu-g++  -I/var/tmp/portage/app-text/xpdf-4.04/work/xpdf-4.04 -I/var/tmp/portage/app-text/xpdf-4.04/work/xpdf-4.04_build -I/var/tmp/portage/app-text/xpdf-4.04/work/xpdf-4.04/goo -I/var/tmp/portage/app-text/xpdf-4.04/work/xpdf-4.04/fofi -I/var/tmp/portage/app-text/xpdf-4.04/work/xpdf-4.04/splash  -pipe -march=native -fno-diagnostics-color -O2 -fPIC -MD -MT xpdf/CMakeFiles/xpdf_objs.dir/GlobalParams.cc.o -MF xpdf/CMakeFiles/xpdf_objs.dir/GlobalParams.cc.o.d -o xpdf/CMakeFiles/xpdf_objs.dir/GlobalParams.cc.o -c /var/tmp/portage/app-text/xpdf-4.04/work/xpdf-4.04/xpdf/GlobalParams.cc
/var/tmp/portage/app-text/xpdf-4.04/work/xpdf-4.04/xpdf/GlobalParams.cc: In constructor ‘GlobalParams::GlobalParams(const char*)’:
/var/tmp/portage/app-text/xpdf-4.04/work/xpdf-4.04/xpdf/GlobalParams.cc:661:35: error: invalid conversion from ‘const char*’ to ‘char*’ [-fpermissive]
  661 |   if ((paperName = systempapername())) {
      |                    ~~~~~~~~~~~~~~~^~
      |                                   |
      |                                   const char*
```

Do you happen to know at a glance if this is an issue on libpaper's end (as it seems to have compiled before) or if it's xpdf?

Thank you!

**Comment by rrthomas** [comment]
This is a change in libpaper 1.2.4 (and of course libpaper 2). I checked the source (I was unaware of xpdf 4.x! it's not yet packaged in Debian) and the variable is never written as far as I can see, so no problem here, though of course it would be good to patch in xpdf.

**Comment by rrthomas** [comment]
@thesamesam is there anything else for libpaper here? Sounds as though the simplest fix is just for `paperName` to be declared `const char *` in xpdf?

**Comment by thesamesam** [comment]
I'm sorry, I completely forgot! I don't think there's anything for you to do here and I'll make sure it gets reported to xpdf now if it hasn't been already. But thank you for being so helpful!

**Comment by thesamesam** [comment]
I see you changed it back in https://github.com/rrthomas/libpaper/commit/a99a58a83097bc09ce800f3a4107aeec654e4404 - any particular reason?

**Comment by rrthomas** [comment]
I'm sorry, I had forgotten about this issue. I changed it because I realised it was actually wrong: the whole point of `systempapername` is that it `malloc`s its return value. I think I went through a couple of iterations on this: originally, I reimplemented `systempapername` as just a macro `#define`d to `defaultpapername`, then I `malloc`ed the return value (because some callers were trying to `free` it, but didn't change the function prototype; then I realised that the function prototype had to change.
